src/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -1430,8 +1430,10 @@ public static String validateCanal(final Route route, final Collection<Unit> uni
         continue;
       }
       failureMessage = canPassThroughCanal(canalAttachment, units, player, data);
-      if (!failureMessage.isPresent()) {
```

Comment:
13 lines down to 5.  It took about 10 minutes to understand this block of code. It's complex. Please take the time to watch the video, there is an interesting analysis of a filter mess being unfamiliar and complex, and a refactor that shows how it simplifies things. If you learned Java today, and looked at the two side by side, one with 13 lines, 3 return points, nested if statments, vs another is 6 lines with declarative statements that indicate function, it's pretty clear which is simpler.

Getting back to the problem at hand, hacking in a new if conditional is not appropriate. This should be worked in so the code does not grow ever more complex. You've added another return point, which is not good and has added to the complexity.